pages/blog/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { Layout } from '../../layouts/article/layout';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { load } from 'cheerio';
+import { Blog } from '../../components/blog';
+import { sortBy } from 'lodash';
+
+type BlogType = {
+  url: string;
+  title: string;
+  profilePic: string;
+  date: string;
+  name: string;
+};
+
+type Writer = {
+  name: string;
+  username: string;
+};
+
+const getBlogsByUsernames = async (writers: Writer[]) => {
+  return Promise.all(
+    writers.map(async writer => {
+      const baseUrl = 'https://dev.to';
+      let response;
+      try {
+        response = await axios.get(`${baseUrl}/${writer.username}`);
+      } catch (e) {
+        console.log(e.message);
+      }
+
+      if (!response || response.data === undefined) {
+        return;
+      }
+
+      const dom = load(response.data);
+      const nodes = dom('.crayons-story__body');
+
+      const blogs: BlogType[] = [];
+
+      for (let i = 0; i < nodes.length; i++) {
+        const url =
+          nodes[i].children[2].next.children[1].children[1].attribs.href;
+        const date =
+          nodes[i].children[0].next.children[0].next.children[1].next.next
+            .children[2].next.children[0].attribs.datetime;
+        const profilePic =
+          nodes[i].children[1].children[1].children[2].parent.children[1]
+            .children[1].children[1].attribs.src;
+        let title = nodes[i].children[2].next.children[1].children[1]
+          .children[0].data as string;
+
+        if (title) {
+          title = title.trim();
+        }
+
+        blogs.push({ url, date, profilePic, title, name: writer.name });
+      }
+      return blogs;
+    })
+  );
+};
+
+const Header = () => (
+  <section className="header-main">
+    <h1>Blogi</h1>
+  </section>
+);
+
+const Page: React.FC = () => {
+  const writers = [
+    { name: 'Riku Rouvila', username: 'rikurouvila' },
+    { name: 'Hannes Aaltonen', username: 'haalto' }
+  ];
+  const [blogs, setBlogs] = useState<BlogType[] | null>(null);
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedBlogs = (await getBlogsByUsernames(writers)) as BlogType[][];
+      const sortedBlogs = sortBy(fetchedBlogs.flat(), 'date').reverse();
+      setBlogs(sortedBlogs);
```

Comment:
maybe this could `return sortedBlogs` instead of `setBlogs(sortedBlogs)`.

then the line 84 could be used as:
```
fetchData().then(data => setBlogs(data))
```

then ESLint shouldn't cry either